src/spikeinterface/core/waveform_extractor.py

```diff
@@ -1861,8 +1862,20 @@ def __init__(self, waveform_extractor):
         # register
         self.waveform_extractor._loaded_extensions[self.extension_name] = self
 
+    @property
+    def waveform_extractor(self):
+        # Important : to avoid that WaveformExtractor reference a BaseWaveformExtractorExtension
+        # and BaseWaveformExtractorExtension reference a WaveformExtractor
+        # we need a weakref. Otherwise the garbage collecor is not working properly
+        # and so the WaveformExtractor + its recordsing are still alive even after deleting explicitly
+        # the WaveformExtractor which make impossible to delete folder!
+        we = self._waveform_extractor()
+        if we is None:
+            raise ValueError(f"The extension {self.extension_name} has lost its WaveformExtractor")
```

Comment:
This sounds like a very annoying bug. Thanks for findig it out.

Can we give instructions on what to do if this comes to happen. That is, can the assertion error indicate how the problem might be mitgated or solved? Is this possible?

Reply:
Normaly this will never happen.
This would mean deletign the we without deleting the extention which is inside in a list.
But who knows. I put this error as a reminder in case it would happen.

We shoudl rewrite a new postprocessing object in 1 or 2 month so all this would desappear.